src/routers/alpha-router/alpha-router.ts

```diff
@@ -300,6 +314,11 @@ export type AlphaRouterConfig = {
    * This parameter should always be false. It is only included for testing purposes.
    */
   forceCrossProtocol: boolean;
+  /**
+   * Prevent the alpha router from considering mixedRoutes as a valid swap.
+   * Default will be falsy.
+   */
+  disableMixedRoutesConsideration?: boolean;
```

Comment:
I was debating adding a `forceMixedRoutes` key here which would force the SOR to return a mixedRoute quote. Would be helpful for testing but not much else. Any preference to including it or not?

Reply:
TODO: add logic for passing in Protocol.MIXED to protocols in alphaRouter config forces route through mixedRoutes

Reply:
just use protocols flag instead

Reply:
one note i think that even if Protocol.MIXED is not specified (as in its just v3 / v2) we should still search for mixed but if mixed is the only flag then just look at mixed

Reply:
Here's what I have implemented:
- if no protocols are specified, we consider V3, V2 like normal, and MIXED only if tradeType is exactInput and we're not disabling it explicitly via config
- else, if we specify MIXED in the protocols array, we will consider them only if the chain supports V2 liquidity and tradeType is correct and config is not set.

Thus, we support the cases where it's `[Protocol.MIXED], [Protocol.V3, Protocol.MIXED]`, etc. However, I'm curious if we want to route mixed if we explicitly do not add V2 into the array.

Reply:
Gotcha, sounds good - what about the case where no protocols are specified? Should we consider all `v2,v3,mixed`?